packages/manager/src/features/CloudPulse/Utils/FilterBuilder.ts

```diff
@@ -293,7 +329,10 @@ export const checkIfAllMandatoryFiltersAreSelected = (
     return false;
   }
 
-  return serviceTypeConfig.filters.every((filter) => {
+  const mandatoryFilters = serviceTypeConfig.filters.filter(
```

Comment:
The complexities are getting hit, can we move something outside like above and save some time